### PR TITLE
Handle FirstTimeUser, ReturnUser userflows

### PR DESCRIPTION
**Pull request by madim**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -76,4 +76,7 @@ dependencies {
     implementation 'com.github.TCNCoalition:tcn-client-android:0.0.1'
 
     implementation 'pub.devrel:easypermissions:3.0.0'
+
+    implementation "org.koin:koin-android:2.1.5"
+    implementation "org.koin:koin-androidx-viewmodel:2.1.5"
```
Comment (madim) [inline app/build.gradle:81]: added Koin as a dependency injection framework

File: app/src/main/java/org/covidwatch/android/di/AppModule.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package org.covidwatch.android.di
```
Comment (nkumarcc) [inline app/src/main/java/org/covidwatch/android/di/AppModule.kt:1]: What's happening in this file?

File: app/src/main/java/org/covidwatch/android/presentation/SetupBluetoothFragment.kt [hunk 3]
```diff
@@ -55,6 +59,11 @@ class SetupBluetoothFragment : Fragment(R.layout.fragment_setup_bluetooth),
     }
 
     override fun onPermissionsGranted(requestCode: Int, perms: MutableList<String>) {
-        findNavController().navigate(R.id.homeFragment)
+        permissionGranted()
+    }
+
+    private fun permissionGranted() {
+        userFlowRepository.updateSetupUserFlow()
+        findNavController().popBackStack(R.id.homeFragment, false)
```
Comment (nkumarcc) [inline app/src/main/java/org/covidwatch/android/presentation/SetupBluetoothFragment.kt:67]: Does this make it so we won't come back here with the back button?

File: app/src/main/java/org/covidwatch/android/presentation/SplashFragment.kt [hunk 4]
```diff
@@ -14,6 +15,17 @@ class SplashFragment : Fragment() {
     private var _binding: FragmentSplashBinding? = null
     private val binding get() = _binding!!
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val onBackPressed: OnBackPressedCallback = object : OnBackPressedCallback(true) {
```
Comment (nkumarcc) [inline app/src/main/java/org/covidwatch/android/presentation/SplashFragment.kt:21]: What does this do?

File: app/src/main/java/org/covidwatch/android/presentation/HomeFragment.kt [hunk 5]
```diff
@@ -6,16 +6,25 @@ import android.view.LayoutInflater
...
+import org.covidwatch.android.domain.UserFlow
+import org.covidwatch.android.presentation.home.HomeViewModel
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class HomeFragment : Fragment(R.layout.fragment_home) {
 
     private var _binding: FragmentHomeBinding? = null
     private val binding get() = _binding!!
 
+    private val homeViewModel: HomeViewModel by viewModel()
```
Comment (nkumarcc) [inline app/src/main/java/org/covidwatch/android/presentation/HomeFragment.kt:26]: Do you need to pass the repo? Or does Koin help with this?

File: app/src/main/res/navigation/navigation_temp.xml [hunk 6]
```diff
@@ -2,8 +2,8 @@
 <navigation xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
-    android:id="@+id/navigation_temmp"
-    app:startDestination="@id/splashFragment">
+    android:id="@+id/navigation_temp"
```
Comment (nkumarcc) [inline app/src/main/res/navigation/navigation_temp.xml:5]: Should we rename this?

**Review by nkumarcc — APPROVED** [review]
Some nits and understand q's, otherwise looks great to me

File: app/src/main/java/org/covidwatch/android/di/AppModule.kt (hunk 2 above)
Comment (madim) [reply app/src/main/java/org/covidwatch/android/di/AppModule.kt:1]: this is where we'll declare some of our dependencies so we can later access them via the constructor or injections

File: app/src/main/java/org/covidwatch/android/presentation/SetupBluetoothFragment.kt (hunk 3 above)
Comment (madim) [reply app/src/main/java/org/covidwatch/android/presentation/SetupBluetoothFragment.kt:67]: yep :) here we are popping fragment back stack until we hit `homeFragment`. `false` here means not inclusive

File: app/src/main/java/org/covidwatch/android/presentation/SplashFragment.kt (hunk 4 above)
Comment (madim) [reply app/src/main/java/org/covidwatch/android/presentation/SplashFragment.kt:21]: this is for the case where user is in splash screen, and he hits back button (we need to close the app in this case). without this we'll go back to HomeFragment as this is our new starting destination in nav-graph :)

File: app/src/main/java/org/covidwatch/android/presentation/HomeFragment.kt (hunk 5 above)
Comment (madim) [reply app/src/main/java/org/covidwatch/android/presentation/HomeFragment.kt:26]: koin does it for us :)

File: app/src/main/res/navigation/navigation_temp.xml (hunk 6 above)
Comment (madim) [reply app/src/main/res/navigation/navigation_temp.xml:5]: will do so in "cleanup" PR, where I will delete unused resources and files

**Pull request merged**